### PR TITLE
chore(ci): update dependencies and use golangci-lint-action

### PR DESCRIPTION
**Pull request by mmorel-35**
Hi ! I have updated the dependencies with dependabot, setup a cache for the dependencies (actions/cache).
I also moved the job of golangci as an action.

### CODE REVIEW

**Comment by chrischdi** [comment]
Wow, that's a great number of contributions! Thank you :-) 

/lgtm :-)

Let's wait for the actions

**Comment by mmorel-35** [comment]
It failed because I first removed the local golangci script and then changed my mind but in the process the Ugo of the file changed. If you want agree the script can be remove as you can use it directly in your Ide : https://golangci-lint.run/usage/integrations/

File: Makefile [hunk 1]
```diff
@@ -32,14 +32,11 @@ push:
...
 
-test: vet fmtcheck spdxcheck lint
+test: vet fmtcheck spdxcheck
 
 vet:
 	go vet ./...
 
-lint:
-	@hack/check_golangci-lint.sh
-
```
Comment (chrischdi) [inline Makefile:42]: I'd prefer to keep the shell script for now and just remove it from `make all` via `make test` in line 35.

OK for me if it is broken now, I would like to fix it in a follow-up PR then.

The reason for this: I think it is comfortable to have a working command which replicates the ci command and version. We can't enforce someones IDE's :-)

**Comment by mmorel-35** [comment]
I have restored the script but Windows seems to be causing trouble with the filemodes

**Comment by chrischdi** [comment]
> I have restored the script but Windows seems to be causing trouble with the filemodes

Hmm ok, I've never tested all this on windows as I'm developing on Linux. What kind of issue do you see? May it also be git related?

**Comment by mmorel-35** [comment]
The issue is that windows doesn't handle permissions correctly so the ugo stays 644 instead of 755.
Can you do this modification ? You shall be allowed as a maintainer.

**Comment by chrischdi** [comment]
I'm sorry but I'm not able to push to your repository at https://github.com/mmorel-35/kosmoo .

However, you should be able to disable it by
`git config core.fileMode false`

And maybe reset the change by creating a reverse patch:
```
git diff -p -R --no-ext-diff --no-color \
    | grep -E "^(diff|(old|new) mode)" --color=never  \
    | git apply
```

**Pull request closed without merging**